### PR TITLE
Fixes #8717 feat(nimbus): Add release date field to Audience page

### LINKED ISSUE #8717: Add release date to Audience page
Add a release date picker on the Audience page

See notes here: [https://docs.google.com/document/d/1nC_bcAQuGdo8T2ck3v-6HdT9olFffpFoct63CPzbU-w/edit](https://docs.google.com/document/d/1nC_bcAQuGdo8T2ck3v-6HdT9olFffpFoct63CPzbU-w/edit|smart-link)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3404)

### PR DESCRIPTION
**Pull request by eliserichards**
Because

- We want users with First Run experiments to be able to add a release date

This commit

- Adds the UI to the Audience page that allows a user to add a release date
- Hooks up the serializer warnings from #8888 to guarantee that the release date is only used for first run experiments
- Adds an question mark info icon (?) to the name with a description and link to whattrainisit.com so they can quickly find their date

----
image

image

https://github.com/mozilla/experimenter/assets/43795363/60d9495d-91b1-4fcd-a64e-41bb565b1c48

image

### CODE REVIEW

**Comment by eliserichards** [comment]
With regards to the positioning of the field on the Audience page, I checked with Shell and this is her preference for where it should go :)

**Review by yashikakhurana — COMMENTED** [review]
@eliserichards I think if the user selects the release date, we should also show that on the summary page too, what do you think?

**Comment by eliserichards** [comment]
there's already a ticket for that 😄 #8718

**Review by yashikakhurana — COMMENTED** [review]
so I tested it locally, it works perfectly fine, but one thing I will say, we only give options for the first run to mobile applications, so I think we should only show this release date if the type of application is mobile, as we do with a first run check box, we hide the box for non-mobile applications

File: experimenter/experimenter/experiments/tests/api/v5/test_mutations.py [hunk 1]
```diff
@@ -1363,6 +1363,39 @@ def test_update_experiment_branches_with_feature_configs_error(self):
...
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        content = json.loads(response.content)
+        result = content["data"]["updateExperiment"]
+        self.assertEqual(result["message"], "success")
+
+        experiment = NimbusExperiment.objects.get()
+        self.assertEqual(experiment.is_first_run, True)
+        self.assertEqual(experiment.proposed_release_date, datetime.date(2023, 12, 12))
```
Comment (yashikakhurana) [inline experimenter/experimenter/experiments/tests/api/v5/test_mutations.py:1397]: 🎉

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx [hunk 2]
```diff
@@ -33,6 +33,18 @@ import {
...
+  beforeEach(() => {
+    mockWindowOpen = jest.fn();
+    global.window.open = mockWindowOpen;
+  });
+
+  afterEach(() => {
+    global.window.open = origWindowOpen;
+  });
+
   it("renders without error", async () => {
```
Comment (yashikakhurana) [inline experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:48]: oh why these changes? 🤔

Comment (eliserichards) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:48]: I'm using `window.open` so that it opens a new tab with [whattrainisit](whattrainisitnow.com/), so this sets up a mock for window.open :D

**Review by yashikakhurana — APPROVED** [review]
sure hiding the release date can be done in a separate ticket, let's make @shell1 happy :)

**Comment by eliserichards** [comment]
https://mozilla-hub.atlassian.net/browse/EXP-3553 🎉
